lib/seahorn/BvOpSem2.cc

```diff
@@ -579,59 +581,85 @@ class OpSemVisitor : public InstVisitor<OpSemVisitor>, OpSemVisitorBase {
     setValue(*CS.getInstruction(), res);
   }
 
-  void visitAssertIf(CallSite CS) {
-    ScopedStats __stats__("opsem.assert.if");
-    // NOTE: assert.if.not is not supported
+  void visitAssert(Expr ante, Expr conseq, const DebugLoc &dloc) {
+    ScopedStats __stats__("opsem.assert");
     if (VacuityCheckOpt == VacCheckOptions::NONE) {
       return;
     }
-    auto I = CS.getInstruction();
-    Expr ante = lookup(*CS.getArgument(0));
+    // The solving is done incrementally. We only
+    // reset the solver once per assert instruction.
+    // We then add expressions incrementally.
+    // This works because this check never needs to
+    // remove an expression from the solver.
+    m_ctx.resetSolver();
     Stats::resume("opsem.vacuity");
     tribool anteRes = solveWithConstraints(ante);
     Stats::stop("opsem.vacuity");
     // if ante is unsat then report false and bail out
     if (!anteRes) {
-      LOG("opsem", ERR << "Antecedent is unsat/unknown: " << *I << "\n");
+      LOG(
+          "opsem",
+          if (dloc) {
+            ERR << "Antecedent is unsat/unknown: "
+                << "[" << (*dloc).getFilename() << ":" << dloc.getLine() << "]";
+          } else { ERR << "Antecedent is unsat/unknown"; });
       return; // return early since conseq is unreachable
+    } else {
+      LOG(
+          "opsem",
+          if (dloc) {
+            INFO << "Vacuity passed:"
+                 << "[" << (*dloc).getFilename() << ":" << dloc.getLine()
+                 << "]";
+          } else { INFO << "Vacuity passed"; });
     }
-    LOG("opsem", INFO << "Vacuity passed: " << *I << "\n";);
     if (VacuityCheckOpt == VacCheckOptions::ANTE) {
       return; // don't need to process conseq
     }
-    Expr conseq = lookup(*CS.getArgument(1));
-    Expr e = boolop::land(ante, boolop::lneg(conseq));
+    Expr e = boolop::lneg(conseq);
     Stats::resume("opsem.incbmc");
     tribool conseqRes = solveWithConstraints(e);
     Stats::stop("opsem.incbmc");
     if (conseqRes) {
-      LOG("opsem", ERR << "Consequent is sat/unknown: " << *I << "\n");
+      LOG(
+          "opsem",
+          if (dloc) {
+            ERR << "Consequent is sat/unknown: "
```

Comment:
this is only reachable when Sat but prints that might be unknown

Reply:
good point. fixed.